### PR TITLE
Made form navigation language-direction aware

### LINKED ISSUE #5486: Investigate navigation buttons for LTR languages
Can we make sure the < and > arrows stay in the same orientation?

Can we change the flip animation direction?

https://forum.getodk.org/t/when-using-a-rtl-language-in-collect-arrows-for-back-and-next-buttons-point-in-the-wrong-direction/40659

Once we figure out what's possible and rough level of effort we can morph this issue into actions to take.

### LINKED ISSUE #5555: RTL Select widgets text aligment
#### ODK Collect version
the store version 2023.1.2, the master version 1f9155c7f1f6c1d91532993fd45cc9b68270124a
#### Android version
10, 13
#### Device used
Redmi 9T, Galaxy A23
#### Problem description
When RTL language is applied in the app in Select widgets it looks as if radio buttons are aligned to the right, but the text is aligned to the left. The screenshot shows selects with LTR and RTL languages.

ltrRTLselects

#### Steps to reproduce the problem
1. Go to Settings→ User Interface→ Language and choose a RTL language.
2. Go to All widgets form.
3. Go to select widgets.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5486 
Closes #5555

#### What has been done to verify that this works as intended?
I've tested form navigation with both LTR and RTL languages.

#### Why is this the best possible solution? Were any other approaches considered?
I don't think there is anything important to discuss here. I've updated form navigation so now it respects the type of language. 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
We need to make sure that form navigation works well in both LTR and RTL languages. This includes using navigation buttons that now should be fixed (<> icons should have proper direction) and also swiping. Swiping should be consistent with how the navigation buttons work. I've also fixed animations so it works well according to the type of language. 

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/formentry/SwipeHandler.kt [hunk 1]
```diff
@@ -90,9 +91,17 @@ class SwipeHandler(context: Context, generalSettings: Settings) {
                 if (diffX > xpixellimit && diffY < ypixellimit || diffX > xpixellimit * 2) {
                     beenSwiped = true
                     if (e1.x > e2.x) {
-                        onSwipe.onSwipeForward()
+                        if (view!!.context.isLTR()) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/formentry/SwipeHandler.kt:94]: This `!!` jumped out at me but it's no worse than what's there now -- line 73 already does it. It's a bit weird that two conditionals above (and below the first uses of `view!!`) have not-null assertions.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
I mentioned I was feeling less like I should be doing review here but this one looks like something I can do. 😊 The behavior looks just right to me, @grzesiek2010, and so much better. I'm really eager to get it to users.

File: collect_app/src/main/java/org/odk/collect/android/formentry/SwipeHandler.kt (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/formentry/SwipeHandler.kt:94]: Yeah, I didn't like it too... I could use one of the extension functions that assert it's not null but that would mean one more nested statement so I don't know what's better. I think for now it's ok as it is.

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 10

Verified cases:
- issue #5486 is no longer reproducing
- issue #5555 is no longer reproducing (RTL in select widgets)
- RTL language set in phone settings and RTL set in Collect settings
- RTL  language set in phone settings and LTR set in Collect settings
- LTR language set in Collect/device
- in settings moving backwards allowed/ not allowed
- appearance quick, repeats
- map buttons in RTL
- progress bars in RTL
- validate upon forward swipe, defer validation until finalized, Audit with constraints forms
- Track Changes form
- don’t keep activities enabled/disabled
- dark, light theme

Other issues connected wih RTL that are also present on the master version will be filed as separate issues.

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 13

**Pull request merged**